exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/ZipkinSpanExporterBuilder.java

```diff
@@ -53,6 +56,22 @@ public ZipkinSpanExporterBuilder setEncoder(BytesEncoder<Span> encoder) {
     return this;
   }
 
+  /**
+   * Sets the Function that is responsible for transforming an OpenTelemetry {@link SpanData} into
+   * an instance of a Zipkin {@link Span}. The default Function is an instance of {@link
+   * OtelToZipkinSpanTransformer} configured with the local IP address at the time of creation.
+   *
+   * @param transformer the Function used to transform a SpanData to a Zipkin Span instance
+   * @return this
+   * @see OtelToZipkinSpanTransformer
+   */
+  public ZipkinSpanExporterBuilder setOtelToZipkinTransformer(
```

Comment:
I don't want to add this because its not defined in the spec, and I'm not aware of any use cases besides the removal of local ip. Any additional filtering of attributes / otel span data would be better performed by creating a `DelegatingSpanData`. Additionally, a motivated user can use the `setEncoder(BytesEncoder<Span>)` to achieve the same affect. 

I'm happy enough splitting the transform logic.

Reply:
I respectfully disagree and think that it's problematic to treat the spec as comprehensive. By that reasoning, we shouldn't have the encoder or read timeout or whatever exposed either. Exposing this was the primary goal behind the PR in the first place -- to be able to limit/change the default transformation (in our specific case, it is to filter out the local IP address from the local endpoint, which is expressly unspecified in the spec). 

I feel like maybe it's an unpopular opinion these days, but I think that good APIs should be agreeable and friendly by providing users with sane defaults that satisfy 98%+ of uses while still remaining flexible to other advanced or unconventional uses.

I did peek at the spec again, and because the zipkin mapping has a lot of field level mapping specifics, I agree that it would probably would be a mistake in this case to allow the user to change the mapping entirely. 

I do like the idea of the `DelegatingSpanData` as a workaround and want to try that in our use case. With that in mind, it probably doesn't even make sense then to allow the `Supplier<Optional<InetAddress>>` to be user configurable and we can hide that at the package too.

Reply:
Ok, well, the root problem is that local IP address is obtained directly by the exporter and is not part of the `SpanData` at all, so a `DelegatingSpanData` doesn't actually help us here. The encoder is out current work around, but it definitely feels hacky and this doesn't feel to me like an encoding problem.

Reply:
Ok, hopefully found a compromise. Removed the transformer from the builder and changed it to only be the `Supplier<InetAddress>`. As talked about in #4676 we can change this in a future PR to be a caching `Supplier` that only fetches periodically, and users that don't want the IP address exposed can override the default while keeping all the other transformation per spec.

Reply:
To be more specific, I don't want our APIs to have multiple ways to accomplish the same thing. If a user wants to filter / enrich `SpanData` before exporting, they should use a `DelegatingSpanData` and wrap `ZipkinSpanExporter` with a `SpanExporter` implementation that makes the changes `SpanData` before delegating on. Or for certain types of enrichment, a `SpanProcessor` implementation could be used as well. 

Allowing full customization of otel `SpanData` to zipkin `Span` gives the user yet another way to accomplish this, which is a worse user experience. As you point out, the local address situation is unique because it's not about transforming otel `SpanData` to zipkin `Span`. It's a field that is unique to the zipkin data model, with no corollary in otel. 

If there are several other similar scenarios in which we need to allow for customizing fields that are unique to the zipkin data model, then exposing full customization via `Function<SpanData, Span>` may make sense, since adding more and more one-off customization options isn't great. However, looking at the zipkin data model and the transformation logic, I think the local address situation is unique - I don't see anything else that couldn't be accomplished with `DelegatingSpanData`.

Reply:
Sure, I get it. This is why I pulled back the change that allowed any arbitrary `Function<SpanData,Span>` (span trasnformer) to be passed to the exporter. Instead, the user can only pass an instance of `OtelToZipkinSpanTransformer` which is `final` and really only affords customization of the ip address (via `Supplier`) used when creating the local endpoint.

I went from most widely flexible to nearly most restrictive. Are you saying this isn't enough?

The only other thing I can think to do is to change the supplier to a booean/flag like `omitIpAddressFromLocalEndpoint` but that defeats the future goal of detecting IP address changes (see comment below)

Reply:
> I went from most widely flexible to nearly most restrictive. Are you saying this isn't enough?

No was just adding context. However, there is no need for `OtelToZipkinSpanTransformer` to be public. At this point it's an implementation detail since `ZipkinSpanExporterBuilder` has a setter for `Supplier<InetAddress>` rather than for `OtelToZipkinSpanTransformer`.

Reply:
OH! Gotcha. I definitely lost track of that among all this (the fact that the builder also took the supplier). I'll go ahead and make the Transformer package private.